packages/frontend/src/ee/features/aiCopilot/components/ChatElements/AiChartVisualization.tsx

```diff
@@ -1,24 +1,122 @@
+import {
+    AiChartType,
+    ChartType,
+    ECHARTS_DEFAULT_COLORS,
+    type AiAgentMessageAssistant,
+    type ApiAiAgentThreadMessageVizQuery,
+} from '@lightdash/common';
 import { Box } from '@mantine-8/core';
-import { type FC } from 'react';
+import { useMemo, useState, type FC } from 'react';
+import { SeriesContextMenu } from '../../../../../components/Explorer/VisualizationCard/SeriesContextMenu';
 import LightdashVisualization from '../../../../../components/LightdashVisualization';
 import VisualizationProvider from '../../../../../components/LightdashVisualization/VisualizationProvider';
-import { type getChartVisualizationFromAiQuery } from '../../utils/getChartVisualizationFromAiQuery';
+import MetricQueryDataProvider from '../../../../../components/MetricQueryData/MetricQueryDataProvider';
+import UnderlyingDataModal from '../../../../../components/MetricQueryData/UnderlyingDataModal';
 
-type Props = {
-    vizData: ReturnType<typeof getChartVisualizationFromAiQuery>;
+import { type EchartSeriesClickEvent } from '../../../../../components/SimpleChart';
+import { type EChartSeries } from '../../../../../hooks/echarts/useEchartsCartesianConfig';
+import useHealth from '../../../../../hooks/health/useHealth';
+import { useOrganization } from '../../../../../hooks/organization/useOrganization';
+import { useExplore } from '../../../../../hooks/useExplore';
+import { type InfiniteQueryResults } from '../../../../../hooks/useQueryResults';
+import { getChartOptionsFromAiAgentThreadMessageVizQuery } from '../../utils/echarts';
+import { getAiAgentChartConfig } from '../../utils/getAiAgentChartConfig';
+
+type Props = ApiAiAgentThreadMessageVizQuery & {
+    vizConfig: AiAgentMessageAssistant['vizConfigOutput'];
+    results: InfiniteQueryResults;
 };
 
-export const AiChartVisualization: FC<Props> = ({ vizData }) => {
-    if (!vizData) return null;
+export const AiChartVisualization: FC<Props> = ({
+    query,
+    results,
+    type,
+    vizConfig,
+}) => {
+    const { data: health } = useHealth();
+    const { data: organization } = useOrganization();
+    const { metricQuery, fields } = query;
+    const tableName = metricQuery?.exploreName;
+    const { data: explore } = useExplore(tableName);
+    const [echartsClickEvent, setEchartsClickEvent] =
+        useState<EchartSeriesClickEvent | null>(null);
+    const [echartSeries, setEchartSeries] = useState<EChartSeries[]>([]);
+
+    const resultsData = useMemo(
+        () => ({
+            ...results,
+            metricQuery,
+            fields,
+        }),
+        [results, metricQuery, fields],
+    );
+
+    const chartOptions = getChartOptionsFromAiAgentThreadMessageVizQuery({
+        config: vizConfig,
+        rows: results.rows,
+        type,
+    });
+    const chartConfig = getAiAgentChartConfig({
+        type: type,
+        chartOptions,
+        metricQuery,
+    });
```

Comment:
can't we move these to the `select` so we get the caching/invalidation benefits?